vmdb/app/models/ems_openstack_infra.rb

```diff
@@ -29,6 +30,10 @@ def ensure_parent_provider
     end
   end
 
+  def destroy_parent_provider
+    self.provider.destroy if self.provider
```

Comment:
This can be reduced to: `provider.try(:destroy)`  This change removes the 2 rubocop warnings as well.